File: Android/app/src/main/java/com/sozolabs/poop_a_day/ui/TileBackground.kt
package com.sozolabs.poop_a_day.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color

@Composable
fun TileBackground(isDarkMode: Boolean = isSystemInDarkTheme()) {
    val tileSize = 160f
    val groutWidth = 5f

    val tileColor1 = if (isDarkMode) Color(0xFF2E3338) else Color(0xFFEBF0F5)
    val tileColor2 = if (isDarkMode) Color(0xFF24292E) else Color(0xFFDCE3EB)
    val groutColor = if (isDarkMode) Color(0xFF141719) else Color(0xFFC7CCD1)

    Canvas(modifier = Modifier.fillMaxSize()) {
        drawRect(color = groutColor, size = size)

        val cols = (size.width / tileSize).toInt() + 1
        val rows = (size.height / tileSize).toInt() + 1

        for (row in 0 until rows) {
            for (col in 0 until cols) {
                val isAlternate = (row + col) % 2 == 0
                val color = if (isAlternate) tileColor1 else tileColor2

                val x = col * tileSize + groutWidth / 2
                val y = row * tileSize + groutWidth / 2

                drawRect(
                    color = color,
                    topLeft = Offset(x, y),
                    size = Size(tileSize - groutWidth, tileSize - groutWidth)
                )
            }
        }
    }
}

File: iOS/Poop-A-Day/Poop-A-Day/TileBackground.swift
import SwiftUI

struct TileBackground: View {
    @Environment(\.colorScheme) var colorScheme

    let tileSize: CGFloat = 60
    let groutWidth: CGFloat = 2

    var tileColor1: Color {
        colorScheme == .dark
            ? Color(red: 0.18, green: 0.20, blue: 0.22)
            : Color(red: 0.92, green: 0.94, blue: 0.96)
    }

    var tileColor2: Color {
        colorScheme == .dark
            ? Color(red: 0.14, green: 0.16, blue: 0.18)
            : Color(red: 0.86, green: 0.89, blue: 0.92)
    }

    var groutColor: Color {
        colorScheme == .dark
            ? Color(red: 0.08, green: 0.09, blue: 0.10)
            : Color(red: 0.78, green: 0.80, blue: 0.82)
    }

    var body: some View {
        Canvas { context, size in
            context.fill(
                Path(CGRect(origin: .zero, size: size)),
                with: .color(groutColor)
            )

            let cols = Int(size.width / tileSize) + 1
            let rows = Int(size.height / tileSize) + 1

            for row in 0..<rows {
                for col in 0..<cols {
                    let isAlternate = (row + col) % 2 == 0
                    let color = isAlternate ? tileColor1 : tileColor2

                    let x = CGFloat(col) * tileSize + groutWidth / 2
                    let y = CGFloat(row) * tileSize + groutWidth / 2
                    let rect = CGRect(
                        x: x,
                        y: y,
                        width: tileSize - groutWidth,
                        height: tileSize - groutWidth
                    )

                    let path = Path(roundedRect: rect, cornerRadius: 1)
                    context.fill(path, with: .color(color))
                }
            }
        }
    }
}
